Crash when attempting to save lineups to filter config:

fmcd opened the pull request:
If lineup data contains non-ASCII characters, ConfigParser (python 2.7)
crashed during .write with error such as
UnicodeEncodeError: 'ascii' codec can't encode character

Added force encoding of the value string to ascii with replace for unicode
for now to workaround the error for now.

fmcd committed:
Crash when attempting to save lineups to filter config:
If lineup data contains non-ASCII characters, ConfigParser (python 2.7)
crashed during .write with error such as
UnicodeEncodeError: 'ascii' codec can't encode character

Added force encoding of the value string to ascii with replace for unicode
for now to workaround the error for now.

astrilchuk commented:
Thanks for the pull request - it appears that I should have used codecs to open the config file as described here: https://pymotw.com/2/ConfigParser/

Do you want to create a new pull request with using codecs to open the file?

fmcd committed:
Fixed opening of filter config file to use utf-8 codec, and update
filter value setting to encode with utf-8.

Filter file now correctly shows Channel names with utf-8, for example
"Discovery en Español"

fmcd commented:
Thanks for the pointer.. I've updated it and tested it is now producing utf8 filter files correctly from sd lineups.

astrilchuk commented:
Sorry to request further changes....

Can you switch to io.open() please?  io.open() is apparently the default open in Python 3.x, is compatible with 2.x, and can be passed the encoding.  Please use the io.open() for reading and writing to the file (for consistencies sake).

Could you also remove the dir(self._config) line?

fmcd committed:
Switched use of codecs to io for python 3 compatibility
Removed dir() debug line left in by mistake

fmcd committed:
Changed open for write to use io.open

fmcd commented:
Hope that should be all done now. Thanks..

astrilchuk commented:
I'm not clear whether io.open should use rb and wb as options for reading and writing.  I noticed you used r and wb.  Did you notice any issues in your testing?

fmcd commented:
Yes.. I changed that at one point because I ran into an error about the encoding somehow not being compatible with binary file mode. I can't reproduce it right now. But as I understand it, python implements the chosen stream encoding on non-binary files, so this would appear to be the most appropriate option anyway - i.e. binary data not compatible with the encoding should be considered invalid.
